src/osdc/Objecter.h

```diff
@@ -1908,6 +1909,8 @@ class Objecter : public md_config_obs_t, public Dispatcher {
   std::map<ceph_tid_t,StatfsOp*> statfs_ops;
   std::map<ceph_tid_t,PoolOp*> pool_ops;
   std::atomic<unsigned> num_homeless_ops{0};
+  std::map<ceph_tid_t, unsigned> op_refs;
+  std::map<ceph_tid_t, std::function<void()> > waiting_for_op_release;
```

Comment:
This looks like it will have a significant impact on performance.

I'm think pondering this problem.  I think there has got to be a better way to avoid the issue...